### DISCUSSION TITLE
Curious where to find the models that smithy generates from

### DISCUSSION THREAD

**Discussion by eighty4** [smithy-lang/smithy-rs#3956] [General]
I see a couple of sts.json and s3.json type files in this repo. Is there a collection somewhere that models all of the AWS APIs? If so what are the JSON files in this repo?

Do the smithy models include the API documentation that you would find when running `help` with the CLI? Is the man page that's available with `aws cloudfront copy-distribution help` a part of the smithy models? The AWS Rust sdk is very well documented so I thought maybe all of that information was colocated.

**Comment by eighty4** [comment]
The reason I ask:

I have a SWC program that parses Lambda ASTs and thought it would be useful to detect imports of AWS sdks and summarized IAM permissions that a Lambda execution role would need. The WWW and CLI API docs all mention what APIs require what IAM permissions. Does that come from the Smithy IDL?

**Reply by landonxjames** [reply]
You can find the models at https://github.com/awslabs/aws-sdk-rust/tree/main/aws-models, and it looks like at least part of the dialog included in the `help` commands (the description) is included in the models.

**Comment by eighty4** [comment]
Thanks for the help. Nested in each command is the verbatim documentation text.
